src/core/a-node.js

```diff
@@ -97,11 +97,16 @@ module.exports = registerElement('a-node', {
         var children;
         var childrenLoaded;
         var self = this;
+        var primitives = AFRAME.primitives.primitives;
 
         if (this.hasLoaded) { return; }
 
         // Default to waiting for all nodes.
-        childFilter = childFilter || function (el) { return el.isNode; };
+        childFilter = childFilter || function (el) {
+          var tagName = el.tagName.toLowerCase();
+          var isEntity = tagName === 'a-entity' || tagName === 'a-node' || primitives[tagName] !== undefined;
+          return el.isNode || isEntity;
```

Comment:
This function should check isNode the very first. If not, only the execute the lowerCase and other checks